feat(core-kit): add AppErrorState component with Material Design 3 error handling

nesibeyurdakul opened the pull request:
# Pull Request

## 📄 Summary

This PR implements the AppErrorState widget component for the core_kit package, providing a comprehensive and reusable error state UI for the application. The component offers semantic error handling with five distinct error types (network, server, notFound, permission, generic) and supports both full-screen and compact display variants.

The implementation follows Material Design 3 guidelines with proper typography, spacing, and theming. It includes named constructors for common error scenarios, customizable action buttons for retry and secondary actions, and comprehensive Dartdoc documentation.

This component addresses the need for consistent error state presentation across the application and provides developers with an easy-to-use, well-tested solution for handling various error scenarios.

---

## 🧩 Affected Module(s)

Mark the modules impacted by this PR:

- [x] Packages (packages/*)
- [x] Widgetbook Kit (widgetbook_kit/)
- [ ] Documentation (docs/)
- [ ] CI / Infra (.github/)

---

## ✅ Checklist

Before submitting, make sure you've completed the following:

- [x] My branch name follows format: <type>/<short-description> (e.g., feat/login-screen)
- [x] My PR title starts with one of the approved types listed above
- [x] My Dart code is formatted (dart format . or flutter format .)
- [x] I ran static analysis (flutter analyze) and resolved warnings
- [x] I ran tests successfully (flutter test)
- [x] I updated / checked pubspec.yaml and pubspec.lock for dependency changes
- [x] For UI changes, I added screenshots and/or updated golden tests (if applicable)
- [x] I linked related issues using keywords like Closes #42
- [x] I ensured this PR has no unrelated changes
- [x] This PR is ready for review and does not include unfinished work

---

## 🔗 Related Issues

Closes #139 

---

## 💬 Additional Notes

✅ What was implemented
5 semantic error types (network, server, notFound, permission, generic)
2 display variants (full-screen, compact)
Named constructors for common scenarios
Action buttons with retry and secondary actions
Comprehensive Dartdoc documentation
🎨 Material Design 3 Compliance
Typography: headlineSmall/titleLarge
Icon sizes: 64dp (full-screen), 48dp (compact)
Proper spacing and colors from theme
📚 Widgetbook Stories
11 interactive examples covering all error types

🧪 Test Coverage
✅ 29 passing, 2 skipped

✅ Verification
Code formatting: ✅ Passed
Static analysis: ✅ No issues
SPDX compliance: ✅ All files
Ready for review

nesibeyurdakul committed:
chore: trigger PR for feat/app-error-state

nesibeyurdakul committed:
fix: update REUSE.toml to cover all Dart source files

nesibeyurdakul committed:
fix(core-kit): replace hardcoded color with theme-based tertiary color and merge develop

nesibeyurdakul committed:
fix: update REUSE.toml to cover all Dart source files

husamettinarabaci reviewed (changes requested):
## Request for Changes

Thank you for the comprehensive implementation of the AppErrorState component! The code quality is excellent and all functional requirements are met. However, there are two housekeeping issues that need to be addressed before merging:

---

### 🚨 **Issue 1: Accidentally Committed Test Output File**

**File:** `packages/core_kit/test_output.txt` (43KB)

**Problem:**
- This file contains verbose Flutter test logs and should not be committed to the repository
- It appears to be a debug artifact accidentally included in the commit

**Required Action:**
```bash
git rm packages/core_kit/test_output.txt
```

**Additional Step:** Please add this pattern to `.gitignore`:
```
# Test output files
test_output*.txt
**/test_output*.txt
```

---

### ⚠️ **Issue 2: Unnecessary REUSE.toml Changes**

**File:** `REUSE.toml`

**Problems:**

1. **Added `test_output*` pattern** (line 26):
   - This pattern was added to handle the accidentally committed test output file
   - Since the file should be deleted, this pattern is unnecessary

2. **Changed `precedence` from `aggregate` to `override`** (line 28):
   - This change is not needed for auto-generated files
   - `aggregate` is the correct precedence for this section

**Required Actions:**

1. **Remove the `test_output*` pattern:**
```toml
path = [
  "**/.dart_tool/**",
  "**/build/**",
  # ... other patterns ...
  "**/*.mocks.dart",
  # ❌ REMOVE THIS LINE: "**/test_output*",
]
```

2. **Revert precedence change:**
```toml
precedence = "aggregate"  # Change back from "override"
```

**Note:** The addition of the Dart source files annotation block (lines 3-8) is actually a good improvement and should be kept:
```toml
# Dart source files
[[annotations]]
path = ["**/*.dart"]
precedence = "aggregate"
SPDX-FileCopyrightText = "2025 hexaTune LLC"
SPDX-License-Identifier = "MIT"
```

---

### ✅ **What's Great About This PR**

- Comprehensive AppErrorState implementation with all required features
- Excellent test coverage (29 passing tests)
- 11 Widgetbook stories (exceeds requirement of 6-8)
- Full Material Design 3 compliance
- Proper SPDX headers on all source files
- Clean, well-documented code

---

### 📋 **Summary**

Please:
1. Delete `packages/core_kit/test_output.txt`
2. Remove `**/test_output*` from REUSE.toml
3. Change `precedence = "override"` back to `precedence = "aggregate"` in the auto-generated files section
4. (Optional but recommended) Add test output patterns to `.gitignore`

Once these housekeeping items are addressed, this PR will be ready to merge! 🚀

nesibeyurdakul committed:
feat(widgetbook): add interactive playground for AppErrorState

- Add interactive playground with comprehensive knobs
- Remove comparison story for cleaner organization
- Remove AppBar from variant stories
- Fix REUSE compliance by removing test_output.txt
- Add test output patterns to .gitignore

nesibeyurdakul committed:
fix(reuse): revert unnecessary REUSE.toml changes

- Remove test_output* pattern from REUSE.toml
- Revert precedence from override to aggregate
- Test output files are already handled by .gitignore

Addresses code review feedback from @husamettinarabaci

husamettinarabaci committed:
fix(hotfix): reuse.toml and .gitignore changed revert

Reverted REUSE.toml and .gitignore changes

husamettinarabaci committed:
fix(hotfix): intertive playground knobs deprecate

Fixed deprecated knobs problem

husamettinarabaci committed:
fix(hotfix): intertive playground knobs deprecate

Fixed deprecated knobs problem

husamettinarabaci committed:
fix(hotfix): intertive playground knobs deprecate

Fixed deprecated knobs problem

husamettinarabaci committed:
Merge branch 'develop' into feat/app-error-state

husamettinarabaci reviewed (approved):
LGTM
